### PR TITLE
Added math scopes to grammar #44

### PR DESCRIPTION
**Pull request by mangecoeur**
Added scopes but highlighting not working. In theory a basic math highlight should be included by including the `#math-syntax` patterns but for some reason this doesn't work - if you have a better understanding of grammars perhaps you can suggest a fix.

### CODE REVIEW

**Comment by Lukasa** [comment]
Hrm, I'm afraid I don't. We may want to leave this open for someone who does to add detail too.

**Comment by Alhadis** [comment]
Nooooooo, you're doing it all wrong. :<

You don't copy+paste from another language grammar to embed its highlighting inside another. You do this:

```coffee
patterns: [include: "scope.name"]
```

So it should just be this:

```coffee
# Math
{
	name: "source.embedded.latex"
	contentName: "markup.math.block.restructuredtext"
	patterns: [include: "text.tex"]
	begin: "^([ \\t]*)(\\.\\.)\\s(math)(::)\\s*$"
	end:   "^(?!\\s*$|\\1[ \\t]+\\S)"
	beginCaptures:
		2: name: "punctuation.definition.directive.restructuredtext"
		3: name: "support.directive.restructuredtext"
		4: name: "punctuation.separator.key-value.restructuredtext"
}
```

This *does* of course require a user have a TeX grammar installed (if they don't, it won't cause any harm: the highlighting just won't show up). If that's a concern, you can always add it as a package dependency in `package.json` (so Atom will install the language grammar too, if it hasn't been installed already):

```json
"package-deps": [
	"language-latex"
]
```

General best practice is to let the user install optional dependencies if they're not going to need them, but that's a design design I'll leave to you. But the bottom line is you *never* try and embed chunks of unrelated languages inside another grammar... it's a one-way ticket to mess hell.

**Comment by mangecoeur** [comment]
Sorry :P I just mimicked the approach from language-pfm since that seemed to work ok.
Will try to update PR with your suggestion and see if it works

**Comment by Alhadis** [comment]
**Tip:** You can also import certain blocks from another grammar by including the name:

patterns: [include: "scope.name#block-id"]

... where `block-id` is a named pattern-block defined in the target grammar's `repository` property. So, to embed reStructuredText's [directive patterns](https://github.com/Lukasa/language-restructuredtext/blob/master/grammars/restructuredtext.cson#L479) inside another grammar, you'd go:

```coffee
patterns: [include: "text.restructuredtext#directives"]
```

This can be helpful if importing an entire grammar is screwing with the rest of the document, and you only need a certain subset of patterns (such as math-highlighting, in this case).

**Comment by mangecoeur** [comment]
Updated the PR to rather include highlighting from Latex grammar. Still works with preview-inline. The remaining issue is that Latex only applies math highlighting inside `begin{}...\end{}` blocks and I'm not sure how to pick the patterns out from that block and apply them inside the :math: block

File: grammars/restructuredtext.cson [hunk 1]
```diff
@@ -477,87 +477,11 @@ repository:
...
-        name: 'punctuation.definition.arguments'
-      }
-  ]
-
 	# Intepreted text: Multiline with role
 	"math-inline":
 		name: "markup.other.command.restructuredtext"
 		contentName: "markup.math.inline.restructuredtext"
-		patterns: [{include: '#math-syntax'}]
+		patterns: [include: "text.tex.latex"]
```
Comment (Alhadis) [inline grammars/restructuredtext.cson]: Try replacing `text.tex.latex` with `text.tex#math`, mate. =) That'll load the [named block](https://github.com/area/language-latex/blob/master/grammars/tex.cson#L148) from the grammar with the `text.tex` scope.

**Comment by mangecoeur** [comment]
Made changes according to @Alhadis comment - highlighting now works!

File: grammars/restructuredtext.cson (hunk 1 above)
Comment (mangecoeur) [inline grammars/restructuredtext.cson]: Thanks mate, works now

**Comment by Alhadis** [comment]
Just a reminder: users will need the [LaTeX language package](https://github.com/area/language-latex) installed in order for the math highlighting to display. If they don't, well, it just looks like uncoloured source... no big deal. Just wanted to make doubly-sure you were aware of that, though. =)

**Comment by Lukasa** [comment]
@Alhadis In recognition of your great work on this repo over the last few months, and because you know substantially more than I do about Atom highlighting packages, I've just sent you an invite to be a repo collaborator. You don't need my oversight: feel free to merge whenever you're happy with something!

FWIW, this LGTM. =)

**Comment by Alhadis** [comment]
Thanks mate! =) Don't worry, I'm pretty good at being responsible, I'm happy to help manage PRs and issues and stuff.

And yeah, erh, I spend *wayyy* too much time working with language grammars, haha. I'm starting to feel it's an unhealthy obsession... thanks!

**Comment by Lukasa** [comment]
My pleasure! I have so much on my plate that this repo ends up way far down my priority list, so it's good to have people who can help take care of it.

**Comment by Alhadis** [comment]
I'm pretty sure the owner of the File-Icons package thought the same thing, heh. =)

I'll look into/merge this soon, I'm just juggling pull requests ATM.

For an unemployed prick, there's no way I should be this busy.

**Comment by Lukasa** [comment]
There's no rush. =)

**Comment by Lukasa** [comment]
Thanks @mangecoeur! :sparkles: :cake: :sparkles:

**Comment by Alhadis** [comment]
@Lukasa Heh, the package is still technically in an alpha state. The release version is at `v0.16.0`.

Think we should finally make that a `v1.0.0`? ;)

**Comment by Lukasa** [comment]
Heh, I don't really mind. =D Version numbering a package with no API is kinda a "gut feel" thing rather than anything else.

**Comment by Alhadis** [comment]
True, but seeing a major release chillin' at `v0` for a perfectly functional package just seems... wrong, heh.

Cutting a release. :)

**Comment by Lukasa** [comment]
Sounds good to me then. =)

**Pull request merged**